### PR TITLE
#2335: fix missing project name in spack ci

### LINKED ISSUE #2335: Fix failing CI for spack package.
Spack workflow on develop failed with this message:
https://dev.azure.com/DARMA-tasking/DARMA/_build/results?buildId=60702&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=1364e02f-4e1e-51cf-2821-fd252a3c0e34&l=10

### PR DESCRIPTION
**Pull request by thearusable**
Fixes #2335

### CODE REVIEW

File: ci/azure/azure-gcc-10-ubuntu-openmpi-spack.yml [hunk 1]
```diff
@@ -1,6 +1,3 @@
-###############################################################################
```
Comment (cz4rs) [inline ci/azure/azure-gcc-10-ubuntu-openmpi-spack.yml:1]: Put a warning that this file is NOT generated automatically.
Do you have any idea how much effort would it take to integrate this with the script / template?

Comment (thearusable) [reply ci/azure/azure-gcc-10-ubuntu-openmpi-spack.yml:1]: I will investigate the feasibility of integrating it seamlessly or adding a warning message accordingly.

**Comment by cz4rs** [comment]
Is it possible to just use this?

```diff
diff --git a/scripts/workflows-azure.ini b/scripts/workflows-azure.ini
index d2c8ace17..9e23e12f2 100644
--- a/scripts/workflows-azure.ini
+++ b/scripts/workflows-azure.ini
@@ -126,6 +126,15 @@ output_name = ci/azure/azure-gcc-10-ubuntu-openmpi.yml
 vt_lb = 0
 vt_tests_num_nodes = 4
 
+[PR-tests-gcc-10-spack-package]
+test_configuration = "gcc-10, ubuntu, no LB, spack"
+compiler_type = gnu
+compiler = gcc-10
+docker_target = "[% linux %]-spack"
+output_name = ci/azure/azure-gcc-10-ubuntu-spack.yml
+vt_lb = 0
+vt_tests_num_nodes = 4
+
 [PR-tests-gcc-11]
 test_configuration = "gcc-11, ubuntu, mpich, trace runtime, coverage"
 compiler_type = gnu
```

We could even drop the `no LB` and `num_nodes` restriction to make it closer to default build :thinking:

**Comment by JacobDomagala** [comment]
Probably worth noting that vt uses older version of spack in `test_spack_package.sh` script (v0.16.3). This is not a problem right now, but could be in the future (latest versions of external dependencies i.e. Kokkos won't be available)

**Comment by thearusable** [comment]
> Is it possible to just use this:
> 
> ```diff
> diff --git a/scripts/workflows-azure.ini b/scripts/workflows-azure.ini
> index d2c8ace17..9e23e12f2 100644
> --- a/scripts/workflows-azure.ini
> +++ b/scripts/workflows-azure.ini
> @@ -126,6 +126,15 @@ output_name = ci/azure/azure-gcc-10-ubuntu-openmpi.yml
>  vt_lb = 0
>  vt_tests_num_nodes = 4
>  
> +[PR-tests-gcc-10-spack-package]
> +test_configuration = "gcc-10, ubuntu, openmpi, no LB, spack"
> +compiler_type = gnu
> +compiler = gcc-10
> +docker_target = "[% linux %]-spack"
> +output_name = ci/azure/azure-gcc-10-ubuntu-openmpi-spack.yml
> +vt_lb = 0
> +vt_tests_num_nodes = 4
> +
>  [PR-tests-gcc-11]
>  test_configuration = "gcc-11, ubuntu, mpich, trace runtime, coverage"
>  compiler_type = gnu
> ```
> 
> We could even drop the `no LB` and `num_nodes` restriction to make it closer to default build 🤔

@cz4rs I integrated it now into the existing azure template. All parameters are the same as in a default build now.

**Review by cz4rs — APPROVED** [review]
Looks good to me!

**Review by JacobDomagala — APPROVED** [review]

**Review by lifflander — APPROVED** [review]
Looks good.

**Pull request merged**